projects/fsastorefrontlib/src/cms-components/agent/contact-agent-form/contact-agent-form.component.ts

```diff
@@ -86,21 +89,26 @@ export class ContactAgentFormComponent implements OnInit, OnDestroy {
 
   submit(): void {
     this.subscription.add(
-      this.csTicketService
-        .createCsTicketForAgent(
-          this.agentId,
-          this.userId,
-          this.collectDataFromContactAgentForm(this.contactAgentForm.value)
-        )
-        .subscribe(data => {
-          if (data) {
-            this.router.go('/');
-            this.globalMessageService.add(
-              'Ticket has been created',
-              GlobalMessageType.MSG_TYPE_CONFIRMATION
-            );
-          }
-        })
+      this.userIdService.getUserId().subscribe(userId => {
```

Comment:
No good 🙂 , subscribe in subscribe. Refactor please

Reply:
Fixed